### ISSUE TITLE
Uncaught exception when type=[null] is used

### ISSUE THREAD

**Issue by rweigel** [tdegrunt/jsonschema#220]
In the following schema,`null` was used in the type array (this is not a valid JSON schema - `"null"` should have been used).  This causes an uncaught exception when attempting to validate in jsonschema 1.1.1 using node v4.2.6:

```
node_modules/jsonschema/lib/attribute.js:50
      return v.id && ('<' + v.id + '>') || (v+'');
              ^
TypeError: Cannot read property 'id' of null
```

```javascript
var json =
{
	"parameters":
		[
			{ 
				"fill": -1e31
			}
		]
}

var schema =
{
		"type": "object",
		"properties": {
			"parameters": {
				"description": "Parameters in dataset",
				"type": "array",
				"items": {
					"type": "object",
					"properties": {
						"fill": {
							"type": ["string",null]
						}
					}
				}
			}
		}
	}

var Validator = require('jsonschema').Validator;
var v = new Validator();
v.validate(json, schema);
```

**Comment by awwright** [comment]
What would be the expected behavior, do you think?
They only thing I can think of doing would be throwing a slightly different
exception.

**Comment by rweigel** [comment]
An "invalid schema" error and an indication of the offending location would be ideal.  Perhaps one could validate the input schema against a meta-schema (https://github.com/json-schema-org/json-schema-spec)?

**Comment by awwright** [comment]
Ok.
Some basic schema validation could be implemented during the scanning phase, probably. I'll note it as a feature request, since technically this is a user error.

**Comment by awwright** [comment]
This should be fixed in 27cc3a5239e1922ce61f2880d5ec96b592e819b2
